lua/lspconfig/configs.lua

```diff
@@ -279,17 +277,14 @@ function configs.__newindex(t, config_name, config_def)
     if client.config._on_attach then
       client.config._on_attach(client, bufnr)
     end
-    if client.config.commands and not vim.tbl_isempty(client.config.commands) then
-      M.commands = vim.tbl_deep_extend('force', M.commands, client.config.commands)
+    if client.config.commands then
+      M.commands = client.config.commands
     end
-    if not M.commands_created and not vim.tbl_isempty(M.commands) then
-      -- Create the module commands
-      util.create_module_commands(config_name, M.commands)
-      M.commands_created = true
+    for _, command_info in pairs(M.commands or {}) do
+      vim.api.nvim_create_user_command(command_info.name, command_info.command, command_info.opts or {})
```

Comment:
I fully agree with this. Very disruptive change that is difficult to motivate.

Reply:
That is a long story. In the original PR, mjlbach wanted create the entire interface for defining per-language-server commands called `user_commands` with identical signature to `nvim_create_user_comman`.

When I opened this PR (still 95% his work), other maintainers first suggested it be just called `commands` instead, which I changed, then it was discussed it was best to deprecate it entirely.

Right now it reflect all of those changes, in a single PR!. What is the suggestion? Keep it as the original PR by mjlbach and then deprecate it in the next nvim release cycle? Or something else?

Reply:
What is the consensus on this? Should I revert it back to as it was?

Reply:
Thanks! I was in the process of writing something similar. 
Will have something ready for review by (my) today evening.

Reply:
I updated the following part:

```
         local option, value = unpack(vim.split(v, "=", { plain = true }))
        option = option.gsub(v, "^%-+", "")
        opts[opts_aliases[option] or option] = value or true
```

This function is worthy of a test case or two (or five).

Reply:
The command table is already being validated in `configs.lua`
so simply this will do
```lua
function M.create_module_commands(module_name, commands)
  for command_name, def in pairs(commands) do
    local opts = {}
    if def.description and def['description'] ~= '' then
      opts.desc = def.description
    end
    api.nvim_create_user_command(command_name, function(...)
      require('lspconfig')[module_name].commands[command_name][1](...)
    end, opts)
  end
end
```
I am not even sure if the `description` key needs to be verified as its also being explicitly validated

Reply:
Wont that suddenly exclude options like:

```lua
MyCommand = {
  function () end,
  ["-nargs=?"],
  ["-complete=..."],
  ["-bang"] = true
}
```

Reply:
I did check through all the default configurations, and they all only provide 
```lua
name = {
function() stuff end,
description = "whatever"
}
```
Let me see what the docs specify, or if those options have been allowed in the past